python/sparkdl/image/image.py

```diff
@@ -71,11 +87,30 @@ def ocvTypes(self):
 
         .. versionadded:: 2.3.0
         """
-
         if self._ocvTypes is None:
-            ctx = SparkContext._active_spark_context
-            self._ocvTypes = dict(ctx._jvm.org.apache.spark.ml.image.ImageSchema.javaOcvTypes())
-        return self._ocvTypes
+            ctx = SparkContext.getOrCreate()
+            self._ocvTypes = [self._OcvType(name=x.name(),
+                                            mode=x.mode(),
+                                            nChannels=x.nChannels(),
+                                            dataType=x.dataType(),
+                                            nptype=self._ocvToNumpyMap[x.dataType()])
+                              for x in ctx._jvm.org.apache.spark.ml.image.ImageSchema.javaOcvTypes()]
+        return [x for x in self._ocvTypes]
```

Comment:
What does this do? Isn't `self._ocvTypes` already a list?

Reply:
The purpose was to return copy of the list so that the private member can not be modified.

Reply:
oic, I usually see `myList[:]` or `list(myList)` to make a copy.

Reply:
Thanks, that's nicer :) 
The members of the list are tuples so shallow copy suffices here.